viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -4,11 +4,29 @@ import { TraceServerConfigService } from '../common/trace-server-config';
 @injectable()
 export class TraceServerServiceImpl implements TraceServerConfigService {
 
-    async startTraceServer(path: string | undefined, port: number | undefined): Promise<void> {
-        spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+    startTraceServer(path: string | undefined, port: number | undefined): Promise<void> {
+        const server = spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+        return new Promise<void>((resolve, reject) => {
+            // If the server provides output on any channel before it exits, consider that a success.
+            server.stdout.on('data', () => resolve());
+            server.stderr.on('data', () => resolve());
+            // If the server exits or errors before it outputs, consider it a failure.
+            server.on('exit', code => reject(code));
```

Comment:
FYI: The `exit` event passes 2 parameters: `code` and `signal`. Only one is set while the other is `null`.